### PR TITLE
JDK-8308987: Update java.lang.Class to use javadoc snippets

### PR DESCRIPTION
**Pull request by jddarcy**
Straightforward doc refactoring to use snippets.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8308987](https://bugs.openjdk.org/browse/JDK-8308987): Update java.lang.Class to use javadoc snippets

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14194/head:pull/14194` \
`$ git checkout pull/14194`

Update a local copy of the PR: \
`$ git checkout pull/14194` \
`$ git pull https://git.openjdk.org/jdk.git pull/14194/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14194`

View PR using the GUI difftool: \
`$ git pr show -t 14194`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14194.diff">https://git.openjdk.org/jdk/pull/14194.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14194#issuecomment-1565684921)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -352,9 +352,9 @@ static String typeVarBounds(TypeVariable<?> typeVar) {
      * runtime {@code Class} descriptor for the class named
      * {@code java.lang.Thread}:
      *
-     * <blockquote>
-     *   {@code Class t = Class.forName("java.lang.Thread")}
-     * </blockquote>
+     * {@snippet lang="java" :
+     *   Class<?> t = Class.forName("java.lang.Thread")
```
Comment (liach) [inline src/java.base/share/classes/java/lang/Class.java:356]: This should have a final semicolon, as it's already a variable declaration and initialization.

File: src/java.base/share/classes/java/lang/Class.java [hunk 2]
```diff
@@ -415,15 +415,15 @@ private static Class<?> forName(String className, Class<?> caller)
...
+     * }
      *
      * is equivalent to:
      *
-     * <blockquote>
-     *  {@code Class.forName("Foo", true, this.getClass().getClassLoader())}
-     * </blockquote>
+     * {@snippet lang="java" :
+     *  Class.forName("Foo", true, this.getClass().getClassLoader())
+     * }
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/lang/Class.java:426]: Not important but there are at least 3 different indentation widths in the update. Does it look okay in the generated javadoc?

Comment (jddarcy) [reply src/java.base/share/classes/java/lang/Class.java:425]: The previous blockquote tag normalized the space -- adjusted the leading whitespace to be consistent.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**